build_runner/lib/src/daemon/daemon_builder.dart

```diff
@@ -59,6 +59,7 @@ class BuildRunnerDaemonBuilder implements DaemonBuilder {
   @override
   Future<void> build(
       Set<BuildTarget> targets, Iterable<WatchEvent> fileChanges) async {
+    var defaultTargets = targets.cast<DefaultBuildTarget>();
```

Comment:
What about marking `targets` as covariant instead and then just overriding the type. This method does not accept anything but a DefaultBuildTarget so that would be give better static checking. 

If not, then I would not call this `defaultTargets`, which sounds like some sort of fallback set of targets instead of the ones that were passed in.

Reply:
Looks like this works well. This requires a change in `package:build_daemon` so I will follow up with that change.